From 87cdebbcad6f4e16ba711227cbbbd70039f88752 Mon Sep 17 00:00:00 2001
From: YunQiang Su <syq@debian.org>
Date: Mon, 7 Sep 2020 09:29:37 +0800
Subject: [PATCH] stage1: re-mkdir /proc instead of umount if it is a symlink

In docker, the TARGET/proc will be a symlink to /proc.
And if the docker instance is called with --privileged, it will umount
the /proc of the whole instance in setup_proc.
---
 debian/changelog | 3 +++
 functions        | 7 ++++++-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/functions b/functions
index 1ac63f7..065320d 100644
--- a/functions
+++ b/functions
@@ -1183,7 +1183,12 @@ setup_proc () {
 		umount_on_exit /dev/shm
 		umount_on_exit /proc
 		umount_on_exit /proc/bus/usb
-		umount "$TARGET/proc" 2>/dev/null || true
+		if [ -L "$TARGET/proc" ];then
+			rm -f $TARGET/proc
+			mkdir $TARGET/proc
+		else
+			umount "$TARGET/proc" 2>/dev/null || true
+		fi
 
 		# some container environment are used at second-stage, it already treats /proc and so on
 		if [ -z "$(ls -A "$TARGET/proc")" ]; then
-- 
GitLab

